content/300-guides/500-other/900-advanced-database-tasks/06-typesafe-raw-sql/01-safeql-gis.mdx

```diff
@@ -0,0 +1,234 @@
+
+If you haven't already, enable the `postgresqlExtensions` Preview feature and add the `postgis` PostgreSQL extension in your Prisma schema:
+
+```prisma highlight=3,8;add
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["postgresqlExtensions"]
+}
+
+datasource db {
+  provider   = "postgresql"
+  url        = env("DATABASE_URL")
+  extensions = [postgis]
+}
+```
+
+<Admonition type="warning">
+
+If you are not using a hosted database provider, you will likely need to install the `postgis` extension. Refer to [PostGIS's docs](http://postgis.net/documentation/getting_started/#installing-postgis) to learn more about how to get started with PostGIS.
+
+If you're using Docker Compose, you can use the following snippet to set up a PostgreSQL database that has PostGIS installed:
+
+```yaml
+version: '3.6'
+services:
+  pgDB:
+    image: postgis/postgis:13-3.1-alpine
+    restart: always
+    ports:
+      - '5432:5432'
+    volumes:
+      - db_data:/var/lib/postgresql/data
+    environment:
+      POSTGRES_PASSWORD: password
+      POSTGRES_DB: geoexample
+volumes:
+  db_data:
+```
+
+</Admonition>
+
+Next, create a migration using the `prisma migrate dev` command to enable the extension. The output of the migration file should look like the following:
+
+```sql
+
+-- CreateExtension
+CREATE EXTENSION IF NOT EXISTS "postgis";
+
+```
+
+Make sure that the migration is applied by running `prisma migrate status`.
+
+## 2. Create a new model that uses a geographic data column
+
+Add a new model with a column with a `geography` data type once the migration is applied. For this guide, we'll use a model called `PointOfInterest`.
+
+```prisma
+model PointOfInterest {
+  id       Int                                   @id @default(autoincrement())
+  name     String
+  location Unsupported("geography(Point, 4326)")
+}
+```
+
+You'll notice that the `location` field uses an `Unsupported` type. This means that we lose a lot of the benefits of Prisma when working with `PointOfInterest`. We'll be using [SafeQL](https://safeql.dev/) to fix this.
+
+Run the `prisma migrate dev` command to create a migration and the table `PointOfInterest` table in your database.
+
+## 3. Integrate SafeQL
+
+SafeQL is easily integrated with Prisma in order to lint `$queryRaw` and `$executeRaw` Prisma calls. You can reference [SafeQL's integration guide](https://safeql.dev/compatibility/prisma.html) or follow the steps below.
+
+### Install the <inlinecode>@ts-safeql/eslint-plugin</inlinecode> npm package
+
+```terminal
+npm install -D @ts-safeql/eslint-plugin
+```
+
+This ESLint plugin is what will allow for queries to be linted.
+
+### Add <inlinecode>@ts-safeql/eslint-plugin</inlinecode> to your ESLint plugins
+
+You should now add `@ts-safeql/eslint-plugin` to your list of ESLint plugins. In our example we are using an `.eslintrc.js` file, but this can be applied to [any way that you configure ESLint](https://eslint.org/docs/latest/use/configure/)
```

Comment:
```suggestion
You should now add `@ts-safeql/eslint-plugin` to your list of ESLint plugins. In our example we are using an `.eslintrc.js` file, but this can be applied to any way that you [configure ESLint](https://eslint.org/docs/latest/use/configure/).
```

Reply:
shorten the hyperlink here otherwise the line break is weird:

<img width="412" alt="image" src="https://github.com/prisma/docs/assets/4058327/0a9474b9-8ea0-4c4d-8e73-c12560448b1f">